doc/source/conf.py

```diff
@@ -301,25 +301,38 @@
 }
 
 
-common_imports = """\
+header = """\
 .. currentmodule:: pandas
 
 .. ipython:: python
    :suppress:
 
+   import csv
+   from datetime import datetime, timedelta
+   import functools
+   import glob
+   import itertools
+   import os
```

Comment:
All those shouldn't be here, should be visibly imported in the places where they are used.

Can you check if we the currently validated `flake8-rst` files (the ones not ignored in `setup.cfg`), removing them raises an error?

That would apply to all the imports except `import pandas as pd` and `import numpy as np`.

Don't fix the errors in this PR if there are more than couple, but if you can show how many failures are without the imports, that would be great.

Reply:
these are also done.